### PR TITLE
Cachito-provided sources should be cumulative,

### PR DESCRIPTION
**Pull request by rcerven**
include remote sources from all base images it is based on

* OSBS-8626

Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 1]
```diff
@@ -116,8 +116,12 @@ def download_sources(self, urls, insecure=False, download_dir=SRPMS_DOWNLOAD_DIR
 
         req_session = get_retrying_requests_session()
         for url in urls:
-            download_url(url, dest_dir, insecure=insecure,
-                         session=req_session)
+            if isinstance(url, dict):
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py]: Please keep the same structure for both codepaths, i.e. always dict

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 2]
```diff
@@ -167,21 +171,44 @@ def assemble_srpm_url(self, base_url, srpm_filename, sign_key=None):
...
-        archives = self.session.listArchives(self.koji_build_id, type='remote-sources')
+        self.log.debug('get remote_urls: %s', koji_build['build_id'])
+        archives = self.session.listArchives(koji_build['build_id'], type='remote-sources')
         self.log.debug('archives: %s', archives)
-        remote_sources_path = self.pathinfo.typedir(self.koji_build, btype='remote-sources')
+        remote_sources_path = self.pathinfo.typedir(koji_build, btype='remote-sources')
         remote_sources_urls = []
 
         for archive in archives:
             if archive['type_name'] == 'tar':
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py]: Please create dict for all types

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 3]
```diff
@@ -167,21 +171,44 @@ def assemble_srpm_url(self, base_url, srpm_filename, sign_key=None):
...
+
+        :return: list, dicts with URL pointing to remote sources
+        """
+        remote_sources_urls = []
+
+        remote_sources_urls.extend(self._get_remote_urls_helper(self.koji_build))
+
+        koji_build = self.koji_build
+
+        while True:
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py]: Maybe
`while 'parent_build_id' in koji_build['extra']['image']:`
to make it more readable?

Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py]: ok

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py]: ok

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 4]
```diff
@@ -116,8 +116,8 @@ def download_sources(self, urls, insecure=False, download_dir=SRPMS_DOWNLOAD_DIR
 
         req_session = get_retrying_requests_session()
         for url in urls:
-            download_url(url, dest_dir, insecure=insecure,
-                         session=req_session)
+            download_url(url['url'], dest_dir, insecure=insecure,
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py]: Maybe we can call it source(s). `url['url']` seems weird. Also docstring for url param is wrong

Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py]: ok

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 5]
```diff
@@ -115,9 +116,9 @@ def download_sources(self, urls, insecure=False, download_dir=SRPMS_DOWNLOAD_DIR
             os.makedirs(dest_dir)
 
         req_session = get_retrying_requests_session()
-        for url in urls:
-            download_url(url, dest_dir, insecure=insecure,
-                         session=req_session)
+        for source in urls:
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py]: I meant `source in sources` :D

Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py]: ok

**Comment by MartinBasti** [comment]
@rcerven can we merge this?

**Comment by rcerven** [comment]
rn updted: - Cachito-provided sources should be cumulative, include remote sources from all base images it is based on

**Pull request merged**